packages/teuchos/parameterlist/test/ParameterList/ParameterList_UnitTests.cpp

```diff
@@ -878,9 +879,17 @@ TEUCHOS_UNIT_TEST( ParameterList, validateParametersAndSetDefaults )
   const ParameterEntry &default_entry = PL_Main.getEntry("Nonlinear Solver");
   TEST_EQUALITY(default_entry.isDefault(), true);
   TEST_EQUALITY(default_entry.isUsed(), false);
+  // Make sure the value is stored as an integer after validation
+  TEST_NOTHROW(Teuchos::any_cast<int>(default_entry.getAny()));
+}
+
+
+TEUCHOS_UNIT_TEST( ParameterList, validateParametersAndSetDefaults_noDefault )
+{
   // Now make sure we have the correct behavior when not using a default value
-  PL_Main = createMainPL();
+  ParameterList PL_Main = createMainPL();
   PL_Main.set("Nonlinear Solver", "Trust Region Based");
+  ParameterList PL_Main_valid = createValidMainPL();
   PL_Main.validateParametersAndSetDefaults(PL_Main_valid);
   const ParameterEntry &entry = PL_Main.getEntry("Nonlinear Solver");
   TEST_EQUALITY(entry.isDefault(), false);
```

Comment:
Should this unit test not also check the type of the "Nonlinear Solver" entry by adding:

```
TEST_NOTHROW(Teuchos::any_cast<int>(default_entry.getAny()));
```

?

That would make it a little more obvious that comparison between these two unit tests.  But I don't know if that is critical or not given other existing unit testing.

Reply:
That's a good idea.  I just pushed the change.